src/sympc/protocol/aby3/aby3.py

```diff
@@ -109,15 +123,36 @@ def truncate(x: MPCTensor, session: Session) -> MPCTensor:
         # RSPointer - public ops, Tensor Pointer - Private ops
         ptr_list = []
         ptr_name = x.share_ptrs[0].__name__
+
+        # TODO:Shoud be concised,lot of branching done,to improve communication efficiency.
+
         if ptr_name == "ReplicatedSharedTensorPointer":
-            ptr_list.append(x.share_ptrs[0].get_shares()[0].get_copy())
-            ptr_list.extend(x.share_ptrs[1].get_copy().shares)
+
+            if ring_size in {2, PRIME_NUMBER}:
```

Comment:
Q: Can't the if from here and the `if` from `elif` be in a single `if...else` outside those?